lib/player.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Player
+  attr_accessor :choice, :player, :count
+  def initialize
+    @choice = nil
+    @player = 'X'
+    @count = 0
+  end
+
+  def switch
+    @count += 1
+    if (@count % 2).zero?
+      @player = 'X'
+    else
+      @count = 1
+      @player = 'O'
+    end
+  end
+
+  def num_choice(game_board)
+    @choice = gets.to_i
+    until game_board[@choice - 1] == ' ' && @choice.between?(1, 9)
+      puts 'Wrong choice!!!!!!!!!!!'
+      puts 'Please input another number between 1-9 and not used already'
+      @choice = gets.to_i
+    end
+    @choice
+  end
```

Comment:
no `puts` nor `gets` nor `print` in any logic files

Reply:
I suggest you to move that method into the Board class. 
There is no reason to keep it in the Player class:
- It doesn't use anything from that class
- The Player class could work without this method

On the other hand, the simple fact that you pass the board, `@board.game_board` from the Board class, as a parameter in that method means that you can have the same method into the Board class  and 
```ruby
def num_choice(choice)
    ...
    until @game_board[choice - 1] == ' ' && choice.between?(1, 9)
      puts 'Wrong choice!!!!!!!!!!!'
      puts 'Please input another number between 1-9 and not used already'
      ...
    end
    ...
  end
```
That is just a mock as an example